### ISSUE TITLE
[Docker Hub] Add ability to rename/delete user accounts

### ISSUE THREAD

**Issue by pkennedyr** [docker/roadmap#44]
**Tell us about your request**
Users should be able to rename or delete their user accounts.

**Which service(s) is this request for?**
Docker Hub

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
See https://github.com/docker/hub-feedback/issues/11

**Are you currently working around the issue?**
NA

**Additional context**
NA

**Comment by ao** [comment]
+1 ∞

**Comment by tianon** [comment]
Just played with Hub's SSO for the first time, and realized that having SSO choose a username _for_ me makes this even more important (IMO). :see_no_evil:

**Comment by makifoxgirl** [comment]
I'd like to be able to change to Docker hub username too if it's possible. Can I contact anyone who can help me in the meantime? Thank you

**Comment by vonFrakas** [comment]
This was asked for in June of 2015 (https://github.com/docker/hub-feedback/issues/11). Seven years later and still no resolution?

**Comment by fscholzspryker** [comment]
Is the thumbs up enough or should we all post +1 to increase prio? :p

**Comment by ThibaudMoretto** [comment]
+1

**Comment by carchobe** [comment]
+1 please

**Comment by morganh83** [comment]
+1

**Comment by fmurodov** [comment]
+1

**Comment by aastopher** [comment]
+1

**Comment by daniel-souza** [comment]
+1

**Comment by jebwatson** [comment]
+1

**Comment by tomposmiko** [comment]
+1
No redirect needed.

**Comment by GinoPane** [comment]
+1

**Comment by coricarson** [comment]
+1

**Comment by ratio-dd** [comment]
+1

**Comment by tungnguyen17** [comment]
+1

**Comment by broot02** [comment]
+1

**Comment by mgreen-bishs** [comment]
+1

**Comment by douglasss** [comment]
+1

**Comment by Xaypanya** [comment]
+1

**Comment by kcrtr** [comment]
+1

**Comment by DRV2EGR** [comment]
+1

**Comment by yasinskyi** [comment]
+1

**Comment by ChoiIseungil** [comment]
+1

**Comment by manuh-L** [comment]
**+1**

**Comment by aurkaxi** [comment]
+1

**Comment by platoputhur** [comment]
+1

**Comment by danbrown** [comment]
+1

**Comment by yavorg** [comment]
+1

**Comment by LeKSuS-04** [comment]
+1

**Comment by robocordyceps** [comment]
+1

**Comment by abbyhu2000** [comment]
+1

**Comment by cyanogenic** [comment]
+1

**Comment by mirusky** [comment]
+1

They don't even care about it, as time passed and they didn't even commented about it. 

IMO, they MUST have this ability and should warn user that changing name could compromise push / pull and they must change all scripts, and warn their own users...

**Comment by mbani01** [comment]
+1

**Comment by thaJeztah** [comment]
Hi! I work at Docker (but not on the Hub team, so bear with me on details :see_no_evil:). I was actually asking about this issue last week (week before? need to check my Slack history) as I wondered if we made progress on this, and I know it’s a feature that comes up frequently.

I found some people that worked on this, and they gave me some details;

### The short version:

There has been some progress, but no ETA yet, and some of this depends on services we integrate with.

### The slightly longer version:

The team spent some sprints working on this feature; one of the known complications is situations where _images_ also have to be migrated, so their first goal was to fix the “lower hanging fruit”; accounts that have no content (images) in the namespace. They implemented this feature (allow rename, but only if no content is present, otherwise to contact support). This feature was rolled out in a staging environment, but revealed new challenges for cases where single-sign-on integration is enabled; syncing accounts with those services turned out to be, erm, “challenging” :speak_no_evil:. One of the developers on the team became a victim of this, got locked out of their account, and it took literally months (:grimacing:) to have their account reinstated in those services.

So the “good news” is that the work gave us more insight into the scope of this work, and areas that need to be worked. The bad news is that, while we’re in talks with these providers to work on a solution, we don’t have an ETA yet on when this will be sorted.

TL;DR; This feature definitely remains on the backlog (and knowing myself, I’ll probably continue harassing those working on it :innocent: :joy:), but don’t (yet) know “when”.

**Comment by danbrown** [comment]
Thanks for the information @thaJeztah, I can imagine how complicated this can be as something that was never planned from the beginning (we've been something similar recently 😂)

**Comment by thaJeztah** [comment]
Yes, with complex, distributed systems, things are rarely "easy". More so, doing without interrupting services (half a billion daily pulls? 😬)

> I can imagine how complicated this can be as something that was never planned from the beginning

Big nod.

Fun fact; this ticket was actually created by a former colleague who was Product Manager for Docker Hub. At the time we still had various older systems in service, some of which dated all the way back to "the Docker Index" (the old name of Docker Hub). Those parts were on the roadmap to be replaced "at some point", but also had quite some hard-coded assumptions (who would've thought things could change??), and had to be replaced. The replacement services have been running side-by-side for an extended period of time to verify consistency, and I think we recently were able to decomission the last one of the old services (whoop!). 

So, yes, progress has been made, but (per my earlier comment) still lots of "fun" to address.

**Comment by shencw** [comment]
+1

**Comment by tyranron** [comment]
This very well complies with the [recent announcement of deleting Open Source organizations](https://blog.alexellis.io/docker-is-deleting-open-source-images/). So now, any public organization on Docker Hub cannot preserve its images naming without paying. We can create a separate personal account, but we cannot rename it to the former org name once all the images are transferred.

If you have a public org on Docker Hub with public images, you're doomed to:
- either pay 420 USD / year;
- or lost your public images;
- or switch to another namespace name, or even container registry

**Comment by fguille94** [comment]
+1

**Comment by cakesoft-chaitanya** [comment]
+1

**Comment by joeloliver** [comment]
+1

**Comment by thaJeztah** [comment]
For those leaving a `+1`; please  use the "thumbs up" emoji on the first comment (as Github allows sorting tickets based on that, and it doesn't send out e-mail notifications to everyone on this thread 😅)

**Comment by DSDanielPark** [comment]
+1

**Comment by Isaiah-Gondon** [comment]
+1

**Comment by joonseolee** [comment]
+1

**Comment by logmedaily** [comment]
+1

**Comment by capitantoto** [comment]
Hi there! I know this is very silly of me, but apparently I accidentally set my username identical to my password, so this feature is incredibly important to me 😬 
+1 👍

**Comment by capitantoto** [comment]
> @capitantoto While you cannot change your username (until this issue gets resolved!), you can change your password!

Yes, this is absolutely true, and luckily it's a unique password just relevant to Docker. That said, I went the easy way and having no images to my name, jsut deactivated the account and created a new one (with a new password!). Thanks for the prompt response.

**Comment by artbruno35** [comment]
+1

**Comment by b8kings0ga** [comment]
+1, I can't change email now

**Comment by JensvandeWiel** [comment]
+1
when is this going to be done : (

**Comment by modusCell** [comment]
+1
when?

**Comment by kafle1** [comment]
+1

**Comment by Docteur-RS** [comment]
+1
Yesterday would have been great.

**Comment by carlmagumpara** [comment]
+1

**Comment by AnhDOS** [comment]
+1

**Comment by thaJeztah** [comment]
For those leaving a `+1`, please  use  the Thumbs-up emoji in the top-comment instead, which allows the team reviewing the roadmap to sort requests based on "upvotes".

**Comment by daniel-souza** [comment]
> For those leaving a `+1`, please use the Thumbs-up emoji in the top-comment instead, which allows the team reviewing the roadmap to sort requests based on "upvotes".

However, thumbs-up does not notify me that this issue still relevant to others. I guess relevance is more important where there are more comments than upvotes, because discussion provides engagement. After the idea of adding +1, this issue bacame quickly noticed.

**Comment by kennylajara** [comment]
So, I am leaving a +1 in a comment and thumbs up the first comment. This is an important feature.  +1

**Comment by Vicopem01** [comment]
I always like a constant username across all profiles but typos and unavailability make it hard to stick to that rule.
I hope the devs look into rename/edit usernames possibility.

**Comment by aubreymiguel** [comment]
+1

**Comment by bralbral** [comment]
+1

**Comment by huseyincakirli** [comment]
+1   
The namespace value must be changeable. I didn't know that when creating a user, the username will appear in the namespace and the namespace name cannot be changed. I want to be able to change the namespace.

**Comment by functionaldude** [comment]
+1

**Comment by 100tomer** [comment]
+1

**Comment by lozhu20** [comment]
+1 🙇

**Comment by pakoone** [comment]
+1

**Comment by marianosaldarriaga** [comment]
+1

**Comment by joaquinsoza** [comment]
+1

**Comment by lewis1277** [comment]
+1
surprised this isn't possible yet

**Comment by fogrye** [comment]
+1

**Comment by 321hendrik** [comment]
+1

**Comment by vitaliizghonnik** [comment]
+1 more please about adding that feature.

**Comment by warfl0p** [comment]
+1

**Comment by kkartch0** [comment]
+1

**Comment by daveology** [comment]
It's surprising that this feature request hasn't been resolved after almost a decade. The 2015 issue https://github.com/docker/hub-feedback/issues/11 was closed for this one from 2020, which doesn't make sense to me. Prioritizing some features for the Docker community, such as this one, in the future would be nice. Hopefully it will! Many thanks to the team for the ongoing efforts.

**Comment by 5HT2** [comment]
Yeah @daveology it's kind of unfortunate that this hasn't been implemented yet.

Even like, at worse, a compromise could involve reserving the old username and continuing to support pulls, similar to how github will redirect old repository links to the new one.

**Comment by monalisanpereira** [comment]
+1

**Comment by CarvalhoWesley** [comment]
+1

**Comment by geoeco** [comment]
+1

**Comment by AdrianLagartera** [comment]
+1

**Comment by allanasp** [comment]
+ 1

**Comment by allanasp** [comment]
+1

**Comment by s265925** [comment]
+1

**Comment by TimurKurilov** [comment]
+1

**Comment by elijahomolo** [comment]
+1

**Comment by whoiscarlo** [comment]
I'm genuinely surprised that this feature STILL isn't possible. Since 2015 this has been requested and it's now 2024?!?! I'm sure it's not a "trivial" ask but even Github eventually solve this 😉

To look at all of the leaps and strides we've made in technology and see that most common of requests - renaming a username - is still a struggle for a $100 million dollar company is rather strange, no?

Anyways, the most frustrating thing a company can do is ignore their user base. 

Confronting an issue head on goes such a long way, even if people are upset with the answer. It shows integrity. Hell even pretending to care is better than being ghosted 👻... But what do I know 🙃

**Comment by whoiscarlo** [comment]
I'm genuinely surprised that this feature STILL isn't possible. This feature has been requested since 2015, and it’s now 2024. I'm sure it's not a "trivial" ask but even Github eventually solve this 😉

I get that this might not be an easy change, but it’s a fundamental feature that so many people have asked for.

It would be great to at least have an update on whether this is being considered or if there are any plans to implement it...

**Comment by chad-earthscope** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by sepsol** [comment]
+1

**Comment by Aleksandre1337** [comment]
+1

**Comment by ahmadalkhansa** [comment]
+1

**Comment by AiEvaluation12** [comment]
+1

**Comment by eng-atwijukire** [comment]
Badly need this feature!

**Comment by albalblo** [comment]
Quite a needed feature

**Comment by enchant3dmango** [comment]
+1

**Comment by briiian** [comment]
+1

**Comment by OmarNunezG** [comment]
+1

**Comment by spar** [comment]
+1

**Comment by almas-x** [comment]
+1

**Comment by geoeco** [comment]
+1

**Comment by guptaashwanee** [comment]
It’s been 5 years, unbelievable

**Comment by fredleger** [comment]
> It’s been 5 years, unbelievable

You meant 6 ? 😅
